go/tools/builders/checker_main.go

```diff
@@ -44,39 +43,54 @@ import (
 // do not return an error so as not to unnecessarily interrupt builds.
 func run(args []string) error {
 	archiveFiles := multiFlag{}
+	importMapIn := multiFlag{}
+	filenames := multiFlag{}
 	flags := flag.NewFlagSet("checker", flag.ContinueOnError)
 	flags.Var(&archiveFiles, "archivefile", "Archive file of a direct dependency")
+	flags.Var(&importMapIn, "importmap", "Import maps of a direct dependency")
+	flags.Var(&filenames, "filename", "A source file being compiled")
+	vetTool := flags.String("vet_tool", "", "The vet tool")
+	stdLib := flags.String("stdlib", "", "The directory containing stdlib packages")
+	packageList := flags.String("package_list", "", "The file containing the list of standard library packages")
 	if err := flags.Parse(args); err != nil {
 		log.Println(err)
 		return nil
 	}
-	goroot, ok := os.LookupEnv("GOROOT")
-	if !ok {
-		log.Fatalf("GOROOT not set")
-	}
-	importsToArchives := make(map[string]string)
-	for _, a := range archiveFiles {
-		kv := strings.Split(a, "=")
-		if len(kv) != 2 {
-			continue // sanity check
+
+	if enableVet {
+		vcfgFile, err := makeVetCfg(*packageList, *stdLib, archiveFiles, importMapIn, filenames)
```

Comment:
One problem is that analysis.Finding values are specific by token positions which we cannot accurately extract from vet's command-line output. Another problem is that the analysis.Package that each analysis is given is not enough to construct the vet.cfg file that vet needs to run.

The first problem I can hack around by leaving the token positions blank and adding special cases to the filtering and error-printing logic to account for these empty tokens. Fixing the second second problem is trickier. Even if I create the vet.cfg outside of the analysis-loop, the analysis API doesn't give me a way to hand this file path to a specific analysis.

Another route I explored was special-casing in the main analysis loop so that it runs in parallel with all the other analyses. However, this made the implementation of runAnalyses() and checkAnalysisResults() a lot messier, so I decided against it. This option might be worth revisiting if vet becomes a big bottleneck in the checker program.